server/recipeAPI/spoonacularAPI/SpoonacularRecipeAPI.ts

```diff
@@ -0,0 +1,312 @@
+import axios from 'axios';
+import * as dotenv from 'dotenv';
+import { URLSearchParams } from 'url';
+import IncorrectIDFormat from '../../exceptions/IncorrectIDFormat';
+import NoParameterFound from '../../exceptions/NoParameterFound';
+import ParameterIsNotAllowed from '../../exceptions/ParameterIsNotAllowed';
+import IFoodAPI from '../../foodAPI/IFoodAPI';
+import IFood from '../../serverAPI/model/food/IFood';
+import IInstruction from '../../serverAPI/model/instruction/IInstruction';
+dotenv.config();
+
+import IRecipe from "../../serverAPI/model/recipe/IRecipe";
+import IRecipeAPI from "../IRecipeAPI";
+
+export default class SpoonacularRecipeAPI implements IRecipeAPI {
+    protected foodAPI: IFoodAPI;
+
+    protected apiKey: string;
+    protected host: string;
+
+    protected recipeSearchParameters: Set<string>;
+    protected recipeGetParameters: Set<string>;
+
+    protected dishTypes: Set<string>;
+
+    protected headers: any;
+
+    constructor(foodAPI: IFoodAPI) {
+        this.foodAPI = foodAPI;
+
+        this.apiKey = process.env.SPOONACULAR_API_KEY;
+        this.host = process.env.SPOONACULAR_HOST;
+
+        this.dishTypes = new Set([
+            "main course",
+            "side dish",
+            "dessert",
+            "appetizer",
+            "salad",
+            "bread",
+            "breakfast",
+            "soup",
+            "beverage",
+            "sauce",
+            "drink"
+        ]);
+
+        this.recipeSearchParameters = new Set([
+            'query',
+            'cuisine',
+            'diet',
+            'intolerences',
+            'includeIngredients',
+            'type'
+        ]);
+
+        this.recipeGetParameters = new Set([
+
+        ]);
+    }
+
+    /**
+     * Retrieves collection of partially filled Recipe items that satisfy searching parameters.
+     * 
+     * @param parameters parameters used for searching.
+     * - query - required parameter that defines the name of the Recipe item (partial names are accepted).
+     * - number - optional parameter that defines max number of the results to be returned.
+     * - page - optional parameter that defines page number for pagination.
+     * - intolerence - optional parameter that defines the type of intolerences to be taken in consideration during searching.
```

Comment:
Please ctrl+f and replace all intolerence to intolerance everywhere in all your branches

Reply:
:(